### PR TITLE
Tp execution input size

### PR DESCRIPTION
**Pull request by breznak**
- test case for issue described in #267 
- hotgym improvements
  - can choose which units to benchmark
  - can get individual timings
  - moved to class, separate header, ...

### CODE REVIEW

**Comment by breznak** [comment]
The PR fails as the test is known to fail now. 

- we should test this with `valgrind --cachegrind` to validate the issue is caused by suboptimal caching for such large data. 
- how to improve the above? 
  - more size-compact SDRs
  - compile flags to better hint cache use? 
  - apparently SP has too much of data for the cache, so it cannot be effective. TP still could. 

- counter argument for it being caused by cache only: 
 - TM(conn) still performs well (under same conditions = cache used by SP)

File: src/test/unit/algorithms/Cells4Test.cpp [hunk 1]
```diff
@@ -560,3 +562,13 @@ TEST(Cells4Test, testEqualsOperator) {
     ASSERT_TRUE(cells1 == cells2);
   }
 }
+
+
+TEST(Cells4Test, testPerformanceInputSizeImpact) {
```
Comment (breznak) [inline src/test/unit/algorithms/Cells4Test.cpp]: There's a couple of nice hotgym changes. 
Should I mark this known-failing test as SKIPPED_ and we can merge this? 
The issue would still remain open.

**Comment by breznak** [comment]
Please review the PR, adds a test and modifies Hotgym. The test correctly reproduces the know issue https://github.com/htm-community/nupic.cpp/issues/267 but is commented out so the CI does not always fail.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -360,21 +369,23 @@ add_subdirectory(test)
...
   target_link_libraries(${src_executable_hotgym} 
-        ${core_library}
+    ${core_library}
 		${COMMON_OS_LIBS}
 		)
 else()
   # for Linux,OSx, link with the shared library
   target_link_libraries(${src_executable_hotgym} 
-        ${src_lib_shared}
+    ${src_lib_shared}
```
Comment (dkeeney) [inline src/CMakeLists.txt]: Tabs show askew in diff.  Can you set your editor to replace tabs with spaces, then set tabs at 4 for CMake?

Comment (breznak) [reply src/CMakeLists.txt]: vim started to misbehave, i'll try to fix that. sorry,

**Pull request merged**